packages/datepicker/src/ux-datepicker.ts

```diff
@@ -146,7 +142,13 @@ export class UxDatepicker implements UxInputComponent {
 
     let parseValue;
 
-    parseValue = this.type === 'time' ? moment(this.textboxValue, this.parsers.time) : moment(this.textboxValue);
+    if (this.type.toLowerCase() === 'date') {
+      parseValue = moment(this.textboxValue, this.formatters.date);
+    } else if (this.type.toLowerCase() === 'time') {
+      parseValue = moment(this.textboxValue, this.formatters.time);
+    } else {
+      parseValue = moment(this.textboxValue, this.formatters.datetime);
+    }
```

Comment:
```suggestion
    const pickerType = this.type.toLowerCase();
    if (pickerType === 'date') {
      parseValue = moment(this.textboxValue, this.formatters.date);
    } else if (pickerType === 'time') {
      parseValue = moment(this.textboxValue, this.formatters.time);
    } else {
      parseValue = moment(this.textboxValue, this.formatters.datetime);
    }
```

Reply:
perf difference will be minor, though we don't want to repeatedly call `.toLowerCase()`

Reply:
Yes this can be optimized. I've done it by calling `.toLowerCase()` on `typeChanged()` so that it's really only called once per type change and not for all valueChanged ...